js/ai.js

```diff
@@ -67,6 +67,14 @@ export default {
         this.test = 0;
         this.foundSet = false;
 
+        //have ai check each time it resumes if game should end; copied from board logic
+        if (isDev)  console.log(`ai sets found: ${this.findValidSets().length}`);
+        if (isDev)  console.log(`ai cards left: ${deck.stock.length}`);
+        if (this.findValidSets().length === 0 && deck.stock.length === 0) {
+            setTimeout(() => game.end(), game.delay['add-cards']);
```

Comment:
maybe add a game.delayEnd() function that encapsulates the timeout so that if in the future a different delay is used, it would be consistent across end calls.

Reply:
Upon review and testing, it does seem like this version needs a timeout since it doesn't need to wait for any card actions.  I've removed the timeout entirely in the latest commit, and am going to do another commit to remove the alternate end condition that happens in board.js since I believe checking on each ai Resume will catch all instances on its own.